Deployment scripts rework

bakhtin opened the pull request:
<!-- You will not see HTML commented line in Pull Request body -->
<!-- Optional sections may be omitted. Just remove them or write None -->

<!-- ### Requirements -->
<!-- * Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion. -->
<!-- * All new code must have code coverage above 70% (https://docs.codecov.io/docs/about-code-coverage). -->
<!-- * CircleCI builds must be passed. -->
<!-- * Critical and blocker issues reported by Sorabot must be fixed. -->
<!-- * Branch must be rebased onto base branch (https://soramitsu.atlassian.net/wiki/spaces/IS/pages/11173889/Rebase+and+merge+guide). -->


### Description of the Change
- Deployment scripts (Ansible playbook) are reworked without platform-dependent binaries in it
- Squashed different deployment scenarios into a single role
- Remove older Ansible roles
- Update role's README file

Key generation process is now implemented with a Python script. There is also a Bash script (`config_gen.sh`) to generate Iroha keys and `genesis.block` that is executed from inside an Ansible task.

**UPDATE:** Fixed deployment scripts to support RC2
<!-- We must be able to understand the design of your change from this description. If we can't get a good idea of what the code will be doing from the description here, the pull request may be closed at the maintainers' discretion. -->
<!-- Keep in mind that the maintainer reviewing this PR may not be familiar with or have worked with the code here recently, so please walk us through the concepts. -->

### Benefits
Platform-independent Ansible role that considers multiple deployment scenarios (including local host)
<!-- What benefits will be realized by the code change? -->

### Possible Drawbacks 
None
<!-- What are the possible side-effects or negative impacts of the code change? -->
<!-- If no drawbacks, explicitly mention this (write None) -->

### Usage Examples or Tests *[optional]*
See README.md
<!-- Point reviewers to the test, code example or documentation which shows usage example of this feature -->

neewy commented:
Tried to run it locally on ubuntu server 18.04.1, failed to run. I am checking what is happening at the moment with my environment. It would be nice to have requirements.txt file for python dependencies

neewy commented on deploy/ansible/roles/iroha-docker/README.md:
```diff
@@ -0,0 +1,86 @@
+## Requirements
+  Tested on Ubuntu 16.04, 18.04
+  - Local:
+    - python-3, python3-dev
```
```suggestion
    - python3, python3-dev
```

l4l commented:
dco :(

bakhtin commented:
Updated playbook to deploy on the remote host by default (previously, it was set up to deploy locally). Corrected README file to reflect these changes. @neewy you may want to follow role's README to run the role once more.

sudomann commented:
> There is also a Bash script (config_gen.sh) to generate Iroha keys and genesis.block that is executed from inside an Ansible task.
🥇 
Man, you did my work for me already. I appreciate your efforts!

neewy commented on deploy/ansible/roles/iroha-docker/defaults/main.yml:
```diff
@@ -0,0 +1,62 @@
+# deployments for testing purposes.
+#
+# Default: false
+overlay_network: true
```
false

neewy commented on deploy/ansible/roles/iroha-docker/README.md:
```diff
@@ -0,0 +1,104 @@
+    - python3, python3-dev
+    - PIP modules: future, ansible(>=2.4), sha3(for Python<3.6)
+  - Remote:
+    - Docker (>=17.12)
```
`sudo apt install docker.io`

bakhtin commented on deploy/ansible/roles/iroha-docker/README.md:
```diff
@@ -0,0 +1,104 @@
+
+    Put this file into `../../inventory/` directory.
+
+`cd ../../ && ansible-playbook -b -e 'ansible_ssh_user=ubuntu' -i inventory/iroha.list playbooks/iroha-docker/main.yml`
```
Added an explanation

bakhtin commented on deploy/ansible/roles/iroha-docker/README.md:
````diff
@@ -0,0 +1,104 @@
+    replicas: 4
+    ```
+
+4. Create a playbook:
````
not required, removed
